### PR TITLE
Dont assign distributor address to pickups

### PR DESCRIPTION
**Pull request by sauloperez**
This is one of the two parts of the `Order` model that get on the way of the Spree 2.0 upgrade as stated in https://community.openfoodnetwork.org/t/order-model-step7/979. TL;DR; `#shipping_method` is removed from `Order` in that version.

So far when saving an order we assign the distributor's address to the orders that are picked up (not delivered) but setting the name and phone number from the billing address.

By inspecting the ceckout's UI as well the as the rails views, this is not shown anywhere, so there's no need to assign it in the first place.

**Current behaviour**

ofn_master

**With this PR**

ofn_without_validation
 
We also had the need to refactor shipping methos by introducing `Delivery`, `Pickup` and `deliverty?`.

While reading the codebase we constantly find ourselves "computing" `the actual meaning og require_ship_address` in our heads to understand whether we're talking about a delivery or a pickup. This refactor makes the itent behind all this logic a bit more obvious, which makes the spree upgrade a bit easier.

### CODE REVIEW

**Comment by kirstenalarsen** [comment]
not shown in interface but I think it might affect reports and maybe invoices, order interface etc? I’d be a bit cautious about thinking this has no implications anywhere

**Comment by sauloperez** [comment]
By looking at the uses of `ship_address` I can't see any impact in those sections. We seem to always check if the order has a ship address first (which means checking for pickup or delivery).

**Comment by myriamboure** [comment]
@sauloperez there might be some intereference as well with the issue @ltrls is working on https://github.com/openfoodfoundation/openfoodnetwork/issues/928, he will put his PR in progress on github today so that you can understand what he has done and see together if there is an impact.

File: spec/controllers/checkout_controller_spec.rb [hunk 1]
```diff
@@ -65,22 +65,31 @@
...
+
       spree_post :update, order: {}
     end
 
     it "only clears the ship address with a pickup shipping method" do
-      order.stub_chain(:shipping_method, :andand, :require_ship_address).and_return false
-      order.should_receive(:ship_address=)
-      controller.send(:clear_ship_address)
+      shipping_method = build(:shipping_method)
+      shipping_method_instance = Pickup.new(order)
```
Comment (enricostano) [inline spec/controllers/checkout_controller_spec.rb]: Do we really need this?

Comment (sauloperez) [inline spec/controllers/checkout_controller_spec.rb]: Not really, the var is not used besides as return value of the stub

**Comment by sauloperez** [comment]
@myriamboure there is no interference so far, but since that is still WIP I can't tell for sure.

**Comment by sauloperez** [comment]
We'll address the problems `shipping_address_from_distributor` brings to the Spree upgrade with a smaller change since this didn't inspire much confidence.

**Pull request closed without merging**